File: fibonacci_algos/lcm/LCM.java
import java.util.*;

public class LCM {
  private static long lcm_naive(int a, int b) {
    for (long l = 1; l <= (long) a * b; ++l)
      if (l % a == 0 && l % b == 0)
        return l;

    return (long) a * b;
  }

  private static int gcd(int a, int b) {
    if ( b == 0 || a == b)
      return a;

    return gcd(b, a%b);
  }

  private static long lcm_fast(int a, int b) {
    /*if (a == b)
      return a;*/
    
    //System.out.println(gcd(a, b));
    long multi = (long) a * b;
    //System.out.println(multi);
    return multi / gcd(a, b);
  }

  public static void main(String args[]) {
    Scanner scanner = new Scanner(System.in);
    int a = scanner.nextInt();
    int b = scanner.nextInt();

    System.out.println(lcm_fast(a, b));
  }
}

File: fibonacci_algos/lcm/lcm.py
# Uses python3
import sys

def lcm_naive(a, b):
    for l in range(1, a*b + 1):
        if l % a == 0 and l % b == 0:
            return l

    return a*b

def gcd(a, b):
	if b == 0 or a == b:
		return a

	return gcd(b, a%b)


def lcm_fast(a, b):
	if a == b:
		return a

	return (a * b) // gcd(a, b)

if __name__ == '__main__':
    #input = sys.stdin.read()
    a, b = map(int, input().split())
    print(lcm_fast(a, b))

